[template.bitset.general][vector.bool.pspc] Make subclauses for `reference` nested classes

AlisdairM opened the pull request:
Consistent with other nested classes within the standard library.  This formulation leads to slightly clearer specification for the member functions that now use appropriately qualified names.

jensmaurer reviewed (changes requested):
In general, I like this.

AlisdairM committed:
[template.bitset.general] Break out 'bitset::reference' into its own subclause

This change is consistent with specification of nested classes
elsewhere in the standard, and allows better use of qualified
names in the declaration of member functions in their own
specification.

AlisdairM committed:
[vector.bool.pspc] Break out 'vector<bool>::reference' into its own subclause

This change is consistent with specification of nested classes
elsewhere in the standard, and allows better use of qualified
names in the declaration of member functions in their own
specification.

tkoeppe commented on source/containers.tex:
```diff
@@ -10607,13 +10594,35 @@
 
+\rSec3[vector.bool.reference]{Class \tcode{vector<bool>::reference}}%
 \pnum
 \tcode{reference}
 is a class that simulates a reference to a single bit in the sequence.
```
@AlisdairM Ping?
